joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -936,23 +922,12 @@ controller_interface::CallbackReturn JointTrajectoryController::on_shutdown(
 }
 
 void JointTrajectoryController::publish_state(
-  const JointTrajectoryPoint & desired_state, const JointTrajectoryPoint & current_state,
-  const JointTrajectoryPoint & state_error)
+  const rclcpp::Time & time, const JointTrajectoryPoint & desired_state,
+  const JointTrajectoryPoint & current_state, const JointTrajectoryPoint & state_error)
 {
-  if (state_publisher_period_.seconds() <= 0.0)
-  {
-    return;
-  }
-
-  if (get_node()->now() < (last_state_publish_time_ + state_publisher_period_))
-  {
-    return;
-  }
-
-  if (state_publisher_ && state_publisher_->trylock())
+  if (state_publisher_->trylock())
   {
-    last_state_publish_time_ = get_node()->now();
-    state_publisher_->msg_.header.stamp = last_state_publish_time_;
+    state_publisher_->msg_.header.stamp = time;
```

Comment:
It seems like the only time when it would be useful to change the time in the header from now to a specified time would be in testing. Would it be better to just leave it as is and then create a standard test function that uses this publish state function correctly?

Reply:
This is still good to have in any case. We are getting time from controller manager to have it consistent. So I would keep this as it is sinc this is the control time